Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs

```diff
@@ -237,21 +187,21 @@ private void LibraryUpdateTimerCallback(object state)
                 // Remove dupes in case some were saved multiple times
                 try
                 {
-                    _logger.LogInformation("Emby.Kodi.SyncQueue: Starting Library Sync...");
+                    _logger.LogInformation("Starting Library Sync...");
                     var startTime = DateTime.UtcNow;                    
 
                     var itemsAdded = _itemsAdded.GroupBy(i => i.Id).Select(grp => grp.First()).ToList();
 
                     var itemsRemoved = _itemsRemoved.GroupBy(i => i.Id).Select(grp => grp.First()).ToList();
 
                     var itemsUpdated = _itemsUpdated
-                                        .Where(i => itemsAdded.Where(a => a.Id == i.Id).FirstOrDefault() == null)
+                                        .Where(i => itemsAdded.FirstOrDefault(a => a.Id == i.Id) == null)
                                         .GroupBy(g => g.Id)
                                         .Select(grp => grp.First())
                                         .ToList();
 
-                    Task x = PushChangesToDB(itemsAdded, itemsUpdated, itemsRemoved, cTokenSource.Token);
-                    Task.WaitAll(x);                    
+                    Task pushToDbTask = PushChangesToDB(itemsAdded, itemsUpdated, itemsRemoved, cTokenSource.Token);
+                    Task.WaitAll(pushToDbTask);                    
```

Comment:
I don't know C# well enough, but I would expect to see `await PushChangesToDB(...)` here, why it's not that way?

Reply:
Can't await inside a lock